packages/test-utils/src/utils.ts

```diff
@@ -101,6 +147,59 @@ export const waitForCondition = async (
     })
 }
 
+/**
+ * Run functions and wait conditions to become true by re-evaluating every `retryInterval` milliseconds. Returns a promise created with Promise.all() 
+ * and waitForCondition() calls. Calls the functions after creating the promise.
+ * 
+ * @param operations function(s) to call
+ * @param conditions condition(s) to be evaluated; condition functions should return boolean or Promise<boolean> and have
+ * no side-effects.
+ * @param timeout amount of time in milliseconds to wait for
+ * @param retryInterval how often, in milliseconds, to re-evaluate condition
+ * @param onTimeoutContext evaluated only on timeout. Used to associate human-friendly textual context to error.
+ * @returns {Promise<unknown[]>} resolves immediately if
+ * conditions evaluate to true on a retry attempt within timeout. If timeout
+ * is reached with conditionFn never evaluating to true, rejects.
+ */
+export const runAndWaitForConditions = (
+    operations: () => void | (() => void)[], 
+    conditions: (() => (boolean | Promise<boolean>)) | (() => (boolean | Promise<boolean>)) [],
+    timeout = 5000,
+    retryInterval = 100,
+    onTimeoutContext?: () => string
+): Promise<unknown[]> => {
+
+    let ops: (() => void)[]
+    if (Array.isArray(operations)) {
+        ops = operations
+    } 
+    else {
+        ops = [operations]
+    }
+    
+    let conds: (() => (boolean | Promise<boolean>)) [] 
+    if (Array.isArray(conditions)) {
+        conds = conditions
+    } 
+    else {
+        conds = [conditions]
+    }
+    
+    const promise: Promise<unknown[]> = new Promise( (resolve, reject) => { 
+        Promise.all(conds.map( (c) => { waitForCondition(c, timeout, retryInterval, onTimeoutContext) }))
```

Comment:
Same problems here: `new Promise` not needed, no promises are returned from `conds.map(…)`

Reply:
Thank you for the review! I made the suggested corrections and pushed again.